llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -433,6 +433,19 @@ struct AAAMDAttributesFunction : public AAAMDAttributes {
       indicatePessimisticFixpoint();
       return;
     }
+
+    bool HasAllocaOrASCast = false;
+    for (BasicBlock &BB : *F) {
```

Comment:
isn't there a `instructions(F)`? Then you don't need 2 breaks, no bool, much easier.

Reply:
Done.